server/graphql/common/expenses.ts

```diff
@@ -632,7 +633,7 @@ export async function createExpense(
     const accountHolderName = payoutMethodData?.accountHolderName;
     const legalName = <string>expenseData.fromCollective.legalName;
     if (accountHolderName && legalName && !isAccountHolderNameAndLegalNameMatch(accountHolderName, legalName)) {
-      throw new Error('The legal name should match the bank account holder name');
+      logger.warn('The legal name should match the bank account holder name');
```

Comment:
Let's show more data, that will help us to identify the problematic cases 

```suggestion
      logger.warn('The legal name should match the bank account holder name (${accountHolderName} ≠ ${legalName})');
```